Sources/Orbit/Components/Stepper.swift

```diff
@@ -0,0 +1,262 @@
+import SwiftUI
+
+/// Enables incremental changes of a counter without a direct input.
+///
+/// - Note: [Orbit definition](https://orbit.kiwi/components/interaction/stepper/)
+public struct Stepper: View {
+
+    @Binding var value: Int
+
+    let minValue: Int
+    let maxValue: Int
+    let isEnabled: Bool
+    let style: Style
+
+    @Environment(\.colorScheme) var colorScheme
+    
+    @State var isIncreased = false
+
+    public var body: some View {
+        HStack(alignment: .center, spacing: 0) {
+            decrementButton
+                .accessibility(addTraits: [.isButton])
+
+            valueContainer
+                .frame(minWidth: 40)
+                .accessibility(value: .init("\(value)"))
+
+            incrementButton
+                .accessibility(addTraits: [.isButton])
+        }
+    }
+
+    @ViewBuilder var valueContainer: some View {
+        Text("\(value)")
+            .foregroundColor(colorScheme == .light ? .cloudDark : .cloudLight)
+            .transition(
+                AnyTransition.asymmetric(
+                    insertion: .move(edge: isIncreased ? .top : .bottom),
+                    removal: .move(edge: isIncreased ? .bottom : .top)
+                )
+                .combined(with: .opacity)
+            )
+            .id("\(value)")
+    }
+
+    @ViewBuilder var decrementButton: StepperButton {
+        StepperButton(
+            isIncrement: false,
+            isEnabled: isEnabled ? value > minValue : false,
+            style: style,
+            clickAction: {
+                isIncreased = false
+
+                withAnimation {
+                    value -= 1
+                }
+            }
+        )
+    }
+
+    @ViewBuilder var incrementButton: StepperButton {
+        StepperButton(
+            isIncrement: true,
+            isEnabled: isEnabled ? value < maxValue : false,
+            style: style,
+            clickAction: {
+                isIncreased = true
+
+                withAnimation {
+                    value += 1
+                }
+            }
+        )
+    }
+
+    /// Creates Orbit Stepper component.
+    public init(
+        value: Binding<Int>,
+        minValue: Int,
+        maxValue: Int,
+        isEnabled: Bool = true,
+        style: Style
+    ) {
+        self._value = value
+        self.minValue = minValue
+        self.maxValue = maxValue
+        self.isEnabled = isEnabled
+        self.style = style
+    }
+}
+
+// MARK: - Subtypes
+extension Stepper {
+
+    struct StepperButton: View {
+
+        let isIncrement: Bool
```

Comment:
This only selects which icon is used - I'd probably replace this property with the icon, since this is only accessible in this file, the danger of some random icon being assigned is very low.

Reply:
Actually not only the icon we need, we also need appropriate colour for it set. The logic for it is encapsulated into `StepperButton` and is dependant on both `isIncrement` and `isEnabled`. We can move it out, but in that case the sense of `StepperButton` seems to be lost